rust/src/polars/mod.rs

```diff
@@ -246,11 +251,13 @@ impl ExtractValue for Series {
 
 impl ExtractValue for String {
     fn extract(literal: LiteralValue) -> Fallible<Option<Self>> {
-        Ok(match literal {
-            LiteralValue::Null => None,
-            LiteralValue::String(string) => Some(string.into_string()),
-            _ => return fallible!(FailedFunction, "expected String, found: {:?}", literal),
-        })
+        if literal.is_null() {
+            return Ok(None);
+        }
+        literal
+            .extract_str()
+            .map(|s| Some(s.to_string()))
+            .ok_or_else(|| err!(FailedFunction, "expected String, found: {:?}", literal))
```

Comment:
We're losing the parallelism with `impl ExtractValue for Series`. If this is the best approach, no worries.